lib/Targets/AIEVecToCpp/TranslateAIEVecToCpp.cpp

```diff
@@ -2508,7 +2502,7 @@ void CppEmitter::setMemRefDimParam(Value memref, unsigned index,
                                    std::string parameter) {
   auto p = std::make_pair(memref, index);
   assert(!paramIndexMapper.count(p) && "memref dimension already set");
-  paramIndexMapper[p] = parameter;
+  paramIndexMapper[p] = std::move(parameter);
```

Comment:
Parameter 'parameter' is passed by value and only copied once; consider moving it to avoid unnecessary copies